File: problem/Max_Avg_Subarray_I/findMaxAverage.py
class Solution:
    def findMaxAverage(self, nums, k):

        windowSum = 0

        for i in range(k):
            windowSum += nums[i]
        maxSum = windowSum

        for i in range(k, len(nums)):
            windowSum = windowSum + nums[i] - nums[i - k]
            if windowSum > maxSum:
                maxSum = windowSum
                
        return float(maxSum) / float(k)

File: problem/Max_Avg_Subarray_I/findMaxAverage.java
class Solution {
    public double findMaxAverage(int[] nums, int k) {
        double windowSum = 0;
        double maxSum = 0;

        if (nums.length < k) {
            System.out.println("Invalid: Window size is larger than array.");
            return -1;
        }

        for (int i = 0; i < k; i++) {
            windowSum += nums[i];
        }

        maxSum = windowSum;

        for (int i = k; i < nums.length; i++) {
            windowSum += nums[i] - nums[i - k];
            maxSum = Math.max(maxSum, windowSum);
        }

        return maxSum/k;
    }
}